pages/api/auth/[...nextauth].js

```diff
@@ -0,0 +1,41 @@
+import NextAuth from "next-auth";
+import GithubProvider from "next-auth/providers/github";
+import CredentialsProvider from "next-auth/providers/credentials";
+
+export const authOptions = {
+  // Configure one or more authentication providers
+  providers: [
+    process.env.VERCEL_ENV === "preview"
+      ? CredentialsProvider({
+          name: "credentials",
+          credentials: {
+            username: {
+              label: "Username",
+              type: "text",
+              placeholder: "username",
+            },
+            password: { label: "Password", type: "password" },
+          },
+          async authorize(credentials) {
+            if (
+              credentials.username === "crops" &&
+              credentials.password === "crops"
+            ) {
+              return {
+                name: "New Crop",
+                email: "test@example.com",
+              };
+            } else {
+              return null;
+            }
+          },
+        })
+      : GithubProvider({
+          clientId: process.env.GITHUB_ID,
+          clientSecret: process.env.GITHUB_SECRET,
+        }),
+    // ...add more providers here
+  ],
+};
+
```

Comment:
Just a curiosity: is it possible here to add an error message when the authentication does not work?

Reply:
Good question - I guess this is handled automatically be the authentication provider. But will keep that in mind for the future, maybe we implement something like "Login not possible"